src/transformers/integrations/fbgemm_fp8.py

```diff
@@ -0,0 +1,154 @@
+# coding=utf-8
+# Copyright 2024 NetEase, Inc. and the HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ..utils import is_accelerate_available, is_fbgemm_gpu_available, is_torch_available, logging
+
+
+if is_torch_available():
+    import torch
+    from torch import nn
+
+if is_accelerate_available():
+    from accelerate import init_empty_weights
+
+if is_fbgemm_gpu_available():
+    import fbgemm_gpu.experimental.gen_ai  # noqa: F401
+
+logger = logging.get_logger(__name__)
+
+
+class FbgemmFp8Linear(torch.nn.Module):
+    def __init__(self, in_features, out_features, bias, weight_dtype=torch.float32):
+        super().__init__()
+        self.in_features = in_features
+        self.out_features = out_features
+
+        self.register_buffer("weight", torch.zeros((out_features, in_features), dtype=torch.float8_e4m3fn))
+        self.register_buffer("weight_scale", torch.zeros((out_features, 1), dtype=weight_dtype))
+        self.register_buffer("input_scale_ub", torch.zeros((1), dtype=weight_dtype))
```

Comment:
one other thing - does this have to be in the `state_dict` of the model?

Its also in the `config.json`, so I think it might be a duplicate

Reply:
Agree with you about the global config. We just thought that it will be safer for the future since there is no reason why the activation upper bound should be the same for each layer. cc @OlivierDehaene

Reply:
even if this is different for every layer, I think we should store it in the config. during model creation time, we can read the current layer name, and parse corresponding quantization config. This way we unify the interface of creating a model without quantization, and a model with quantization.

Reply:
> I think this type of thing in the state dict can cause pain

@robertgshaw2-neuralmagic, what do you have in mind? Is it because it is a calibrated value therefore it could be deployment specific?

The bigger problem is see here is that this value now becomes mandatory for fp8 models which was not the case in the past. Since this value is calibrated it is hard to find a sensible default and it's hard to justify to users to generate this value before being able to save their model.

Reply:
Perfect ! Now that everyone agree on setting a non persistant buffer, I will do the change !